nvmeof: use .nvmeof pool for gateway and remove pool field from CRD (#17947)
rook/rook: Documentation/Storage-Configuration/Block-Storage-RBD/nvme-of.md

- **Ceph Version**: Ceph v20 (Tentacle) or later

## Step 1: Create the NVMe-oF Gateway

The `CephNVMeOFGateway` CRD manages the NVMe-oF gateway infrastructure. The operator will automatically create the following resources:

- **Service**: One per gateway instance for service discovery
- **Deployment**: One per gateway instance running the NVMe-oF gateway daemon

The gateway also requires an internal `.nvmeof` pool for its state. This pool is defined as a

> CephBlockPool CR named `builtin-nvmeof` with `spec.name: .nvmeof` (since K8s resource names

Reviewer: Let's simplify the description, no need for all the details.
```suggestion
CephBlockPool CR with `spec.name: .nvmeof`.
```

Author: done

> cannot start with a dot). Both resources are included in the same YAML file.

Reviewer: No need for this detail.
```suggestion
```

Author: done

rook-ceph.nvmeof.csi.ceph.com-nodeplugin-xvt5g              2/2     Running   0          60s
```

> ## Step 3: Create the StorageClass Data Pool

Reviewer: ```suggestion
## Step 3: Create the Data Pool
```

Author: done

rook-ceph.nvmeof.csi.ceph.com-nodeplugin-xvt5g              2/2     Running   0          60s
```

## Step 3: Create the StorageClass Data Pool

> Create a CephBlockPool that will be used by the StorageClass to store RBD images:

Reviewer: Let's simplify, no need for the detail for rbd images.
```suggestion
Create a CephBlockPool that will be used by the StorageClass to store data:
```

Author: done

## High Availability

> The production example (`nvmeof.yaml`) already configures `instances: 2` for high availability.

Reviewer: ```suggestion
The example (`nvmeof.yaml`) configures `instances: 2` for high availability.
```

Author: done

# Delete the NVMe-oF CSI operator resources
kubectl delete -f deploy/examples/csi/nvmeof/csi-operator-nvmeof.yaml

> # Delete the NVMe-oF gateway and its pool

Reviewer: ```suggestion
# Delete the NVMe-oF gateway and its metadata pool
```

Author: done

kubectl delete -f deploy/examples/nvmeof.yaml

> # Delete the StorageClass data pool (optional)

Reviewer: ```suggestion
# Delete the data pool (optional)
```

Author: done
